src/Cpp/1-getting-started/14-HelloCOM/HelloCOM.cpp

```diff
@@ -0,0 +1,330 @@
+    {
+        std::cout << "D3D11: Failed to create RTV from Back Buffer\n";
+        return false;
+    }
+    // This section reads and compiles both the vertex and pixel shaders
+    UINT compileFlags = D3DCOMPILE_ENABLE_STRICTNESS;
+#if !defined(NDEBUG)
+    compileFlags |= D3DCOMPILE_DEBUG;
+#endif
+    ComPtr<ID3D10Blob> vertexShaderBlob = nullptr;
+    ComPtr<ID3D10Blob> pixelShaderBlob = nullptr;
+    ComPtr<ID3D10Blob> errorBlob = nullptr;
+    if (FAILED(D3DCompileFromFile(
+        L"Assets/Shaders/Main.vs.hlsl",
+        nullptr,
+        D3D_COMPILE_STANDARD_FILE_INCLUDE,
+        "main",
+        "vs_5_0",
+        compileFlags,
+        0,
+        &vertexShaderBlob,
+        &errorBlob)))
+    {
+        std::cout << "D3D11: Failed to read vertex shader from file\n";
+        if (errorBlob)
+        {
+            std::cout << "D3D11: With message: " << static_cast<const char*>(errorBlob->GetBufferPointer()) << "\n";
+        }
+        return false;
+    }
+    if (FAILED(D3DCompileFromFile(
+        L"Assets/Shaders/Main.ps.hlsl",
+        nullptr,
+        D3D_COMPILE_STANDARD_FILE_INCLUDE,
+        "main",
+        "ps_5_0",
+        compileFlags,
+        0,
+        &pixelShaderBlob,
+        &errorBlob)))
+    {
+        std::cout << "D3D11: Failed to read pixel shader from file\n";
+        if (errorBlob)
+        {
+            std::cout << "D3D11: With message: " << static_cast<const char*>(errorBlob->GetBufferPointer()) << "\n";
+        }
+        return false;
+    }
+    
+    if (FAILED(_device->CreateVertexShader(
+        vertexShaderBlob->GetBufferPointer(),
+        vertexShaderBlob->GetBufferSize(),
+        nullptr,
+        &_vertexShader)))
+    {
+        std::cout << "D3D11: Failed to compile vertex shader\n";
+        return false;
+    }
+    if (FAILED(_device->CreatePixelShader(
+        pixelShaderBlob->GetBufferPointer(),
+        pixelShaderBlob->GetBufferSize(),
+        nullptr,
+        &_pixelShader)))
+    {
+        std::cout << "D3D11: Failed to compile pixel shader\n";
+        return false;
+    }
+    constexpr D3D11_INPUT_ELEMENT_DESC vertexInputLayoutInfo[] =
+    {
+        { "POSITION", 0, DXGI_FORMAT_R32G32B32_FLOAT, 0, offsetof(VertexFormat, vertex), D3D11_INPUT_PER_VERTEX_DATA, 0},
+        { "COLOR", 0, DXGI_FORMAT_R32G32B32_FLOAT, 0, offsetof(VertexFormat, color), D3D11_INPUT_PER_VERTEX_DATA, 0},
+    };
+    if (FAILED(_device->CreateInputLayout(
+        vertexInputLayoutInfo,
+        _countof(vertexInputLayoutInfo),
+        vertexShaderBlob->GetBufferPointer(),
+        vertexShaderBlob->GetBufferSize(),
+        &_vertexLayout)))
+    {
+        std::cout << "D3D11: Failed to create default vertex input layout\n";
+    }
+    // This section to create the vertex and index buffer of the triangle
+    const VertexFormat vertices[] =
+    {
+        //         Vertex                  Color
+        { {  0.0f,  0.5f, 0.0f }, { 0.25f, 0.39f, 0.19f } },
```

Comment:
I am not sure if we talked about it, or not, we probably should as a discussion/vote, how we want to initialize things, in a more "obvious" fashion like 

`new VertexPositionColor{ ... }` or.. ah that reminds me, I am quite fond of explicitly named VertexFormat constructs... like VertexPositionColor in this very instance. Would you mind renaming `VertexFormat`?  or how do you feel about it?

(You only have so many vertexformat in your game/engine going, i am aware of that, but you still have a few differnt ones to deal with, be it for the FST/Debug Lights/BoundingBoxes etc... (ignoring vertexpulling for now))